cips/cip-X.md

```diff
@@ -0,0 +1,271 @@
+validating off-chain arbitrary messages.
+
+## Abstract
+
+This CIP addresses the need for arbitrary message signing. These messages can be used for application sign-in by
+providing proof of wallet ownership, but their use is not limited to this case. They can also be utilized for signing
+files, text, and more.
+
+There is no requirement for these messages to be posted to the network. That's why a set of off-chain messages is
+defined with specific characteristics that prevent them from being processed as on-chain transactions.
+
+This CIP defines two different messages. The first one allows users to sign any arbitrary data, including files, text,
+objects, and more. The second one provides a way for application sign-in by defining a standard human-readable message
+format that is parameterized by scope, session details, and security mechanisms.
+
+## Motivation
+
+Having the ability to sign messages off-chain has proven to be a fundamental aspect of nearly any blockchain.
+The notion of signing messages off-chain has many added benefits, such as saving on computational costs and reducing
+transaction throughput and overhead. Within the context of the Cosmos ecosystem, some of the major applications of signing such
+data include, but are not limited to, providing a cryptographically secure and verifiable means of proving validator
+identity and possibly associating it with some other framework or organization. Additionally, having the ability to
+sign Cosmos messages with a Ledger or similar HSM device.
+
+## Documentation
+
+This CIP introduces the concept of off-chain messages. Off-chain messages are messages that are not and cannot be
+submitted to the network.
+
+It opens the door to sign-in in dApps and webs and signing of files, text and objects.
+
+Command-line applications will allow users to sign and verify files with commands like:
+
+ + `simd keys sign [file] --from [key_name] > signed_file.json`
+ + `simd keys verify [file]`
+
+
+Websites and applications will be able to prove account ownership with a human-readable message instead of a random challenge.
+An example format might look like this:
+
+```
+Prove account ownership to
+https://myApp.com
+
+  "appDomain": "myApp",
+  "uri": "https://myApp.com/",
+  "signerAddress": "cosmos1hftz5ugqmpg9243xeegsqqav62f8hnywsjr4xr",
+  "nonce": "14368412",
+  "issued-at": "2023-10-10T10:10:10Z"
+```
+
+This approach is an improvement over the current state, as it allows for the signing of arbitrary data like files and
+offers a more user-friendly sign-in process, replacing non-human-readable random challenges.
+
+## Specification
+
+Off-chain signed messages should resemble Cosmos SDK messages but **must not** constitute a valid on-chain transaction.
+`chain-id`, `account_number`, and `sequence` must all be assigned invalid values.
+
+An off-chain transaction follows these rules:
+
+ + the memo must be empty
+ + nonce, sequence number must be equal to 0
+ + chain-id must be equal to “”
```

Comment:
non-empty chain-id might be helpful if you want to bind the signed object to a specific chain